client/js/restaurants_api.js

```diff
@@ -679,6 +662,16 @@ function createNewReservation(restaurant_name) {
         'startTime': reservation_datetime
       };
 
+  var data = {
+    '@type': 'FoodEstablishmentReservation',
+    'partySize': partySize,
+    'reservationFor': {
+      '@type': 'FoodEstablishment',
+      'name': '' + restaurant_name
+    },
+    'startTime': reservation_datetime+'Z'
```

Comment:
This is really strange. I would suggest you use Javascript Date objects.

Even you can use <input type=date> <input type=time> and valueAsDate